src/account.ts

```diff
@@ -0,0 +1,180 @@
+  address = ethjsUtil.stripHexPrefix(address).toLowerCase()
+  const hash = keccak(address).toString('hex')
+  let ret = '0x'
+
+  for (let i = 0; i < address.length; i++) {
+    if (parseInt(hash[i], 16) >= 8) {
+      ret += address[i].toUpperCase()
+    } else {
+      ret += address[i]
+    }
+  }
+
+  return ret
+}
+
+/**
+ * Checks if the address is a valid checksummed address.
+ */
+export const isValidChecksumAddress = function(address: string): boolean {
+  return isValidAddress(address) && toChecksumAddress(address) === address
+}
+
+/**
+ * Generates an address of a newly created contract.
+ * @param from The address which is creating this new address
+ * @param nonce The nonce of the from account
+ */
+export const generateAddress = function(from: Buffer, nonce: Buffer): Buffer {
+  from = toBuffer(from)
+  const nonceBN = new BN(nonce)
+
+  if (nonceBN.isZero()) {
+    // in RLP we want to encode null in the case of zero nonce
+    // read the RLP documentation for an answer if you dare
+    return rlphash([from, null]).slice(-20)
+  }
+
+  // Only take the lower 160bits of the hash
+  return rlphash([from, Buffer.from(nonceBN.toArray())]).slice(-20)
+}
+
+/**
+ * Generates an address for a contract created using CREATE2.
+ * @param from The address which is creating this new address
+ * @param salt A salt
+ * @param initCode The init code of the contract being created
+ */
+export const generateAddress2 = function(
+  from: Buffer | string,
+  salt: Buffer | string,
+  initCode: Buffer | string,
+): Buffer {
+  const fromBuf = toBuffer(from)
+  const saltBuf = toBuffer(salt)
+  const initCodeBuf = toBuffer(initCode)
+
+  assert(fromBuf.length === 20)
+  assert(saltBuf.length === 32)
+
+  const address = keccak256(
+    Buffer.concat([Buffer.from('ff', 'hex'), fromBuf, saltBuf, keccak256(initCodeBuf)]),
+  )
+
+  return address.slice(-20)
+}
+
+/**
+ * Returns true if the supplied address belongs to a precompiled account (Byzantium).
+ */
+export const isPrecompiled = function(address: Buffer | string): boolean {
+  const a = unpad(address)
+  return a.length === 1 && a[0] >= 1 && a[0] <= 8
+}
+
+/**
+ * Checks if the private key satisfies the rules of the curve secp256k1.
+ */
+export const isValidPrivate = function(privateKey: Buffer): boolean {
+  return secp256k1.privateKeyVerify(privateKey)
+}
+
+/**
+ * Checks if the public key satisfies the rules of the curve secp256k1
+ * and the requirements of Ethereum.
+ * @param publicKey The two points of an uncompressed key, unless sanitize is enabled
+ * @param sanitize Accept public keys in other formats
+ */
+export const isValidPublic = function(publicKey: Buffer, sanitize: boolean = false): boolean {
+  if (publicKey.length === 64) {
+    // Convert to SEC1 for secp256k1
+    return secp256k1.publicKeyVerify(Buffer.concat([Buffer.from([4]), publicKey]))
+  }
+
+  if (!sanitize) {
+    return false
+  }
+
+  return secp256k1.publicKeyVerify(publicKey)
+}
```

Comment:
With these two I am unsure, if they rather belong in the `signature` area or here. Just did some context research, both are [used in wallet](https://github.com/ethereumjs/ethereumjs-wallet/blob/830307cb1b4d083a469e796e2c34064a6c948479/src/index.js#L24), I also had some short look at [this](https://hackernoon.com/a-closer-look-at-ethereum-signatures-5784c14abecc) HackerNoon article on Ethereum signature.

Think we can leave here for now, since we are not doing a release yet we can still move if found to be more appropriate in `signature`.